specs/AppLifecycle/Instancing/CustomEventArgs.md

```diff
@@ -0,0 +1,170 @@
+   The modified [AppActivationArguments](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.appactivationarguments) object can be passed into [RedirectActivationToAsync](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.appinstance.redirectactivationtoasync#microsoft-windows-applifecycle-appinstance-redirectactivationtoasync(microsoft-windows-applifecycle-appactivationarguments)).
+
+# API Pages
+
+## AppActivationArguments
+
+A page for [AppActivationArguments](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.appactivationarguments) exists.
+
+The following is added to the [Remarks](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.appactivationarguments#remarks) section.
+
+If an app wants to pass in custom arguments, the app can call `AddCustomArgs(IInspectable newArgs)` on the [AppActivationArguments](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.appactivationarguments)
+object returned from [GetActivatedEventArgs](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.appinstance.getactivatedeventargs).
+
+## AppActivationArguments.Data
+
+A page for [AppActivationArguments.Data](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.appactivationarguments.data#microsoft-windows-applifecycle-appactivationarguments-data) exists.
+
+The following would be added to the table in the [Remarks](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.appactivationarguments.data#remarks) section.
+
+|_ActivationKind_|_DataType_                             |
+|----------------|---------------------------------------|
+| `UserDefined`  |A custom object that is known to the app|
+
+## Microsoft.Windows.AppLifecycle.ExtendedActivationKind
+
+A page for [Microsoft.Windows.AppLifecycle.ExtendedActivationKind](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.extendedactivationkind) exists.
+
+The following would be added to [Fields](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.extendedactivationkind).
+
+|_Name_|_Contract_|_Description_                       |
+|------|----------|------------------------------------|
+|UserDefined| 1   | `Data` consists of a custom object.|
+
+
+## AppActivationArguments.AddCustomArgs method
+
+This is the class for [AppActivationArguments](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.appactivationarguments).  The only change is adding AddCustomArgs
+
+Example:
+
+```c#
+// The rest of AppActivationArguments here.
+
+public void AddCustomArgs(IInspectable newArguments) 
```

Comment:
Can you provide a sample of _using_ and then _consuming_ the custom activation arguments? That is in the entrypoint of the app, show generating a new custom `AppActivationArguments` instance, then using redirection on it.  And then in the entrypoint handling an activation, show looking for the value and poking at it for the supported sub-kinds.